File: Chapter 05/JS/section_5_4_server_monitor/after/core/ports/alertPort.js
/**
 * PRIMARY PORT (Driven).
 * Defines the contract that all infrastructure adapters must follow.
 * This ensures the Core remains agnostic of specific 3rd-party implementations.
 */
class AlertPort {
    /**
     * Sends an alert message to an external destination.
     * @param {string} message - The content of the alert.
     * @throws {Error} If the subclass does not implement this method.
     */
    sendAlert(message) {
        throw new Error("Method 'sendAlert(message)' must be implemented by the Adapter.");
    }
}

module.exports = AlertPort;

File: Chapter 05/Python/section_5_4_server_monitor/after/core/ports/alert_port.py
from abc import ABC, abstractmethod

class AlertPort(ABC):
    """PRIMARY PORT (Driven).
    
    This Abstract Base Class defines the 'Contract' between the Domain 
    and the Infrastructure. It ensures that any adapter you plug 
    into the system is compatible with the Core's expectations.
    """
    
    @abstractmethod
    def send_alert(self, message: str) -> None:
        """Sends an alert message to an external destination.
        
        Args:
            message (str): The content of the alert to be dispatched.
            
        Raises:
            TypeError: If a concrete adapter fails to implement this method.
        """
        pass